dsl/Delegate.cc

```diff
@@ -109,11 +109,13 @@ vector<unique_ptr<ast::Expression>> Delegate::replaceDSL(core::MutableContext ct
         }
         // sig {params(arg0: T.untyped).returns(T.untyped)}
         // def $methodName(*arg0); end
+        ast::MethodDef::ARGS_store args;
+        args.emplace_back(ast::MK::RestArg(loc, ast::MK::Local(loc, core::Names::arg0())));
+        args.emplace_back(std::make_unique<ast::BlockArg>(loc, ast::MK::Local(loc, core::Names::blkArg())));
         methodStubs.push_back(ast::MK::Sig1(loc, ast::MK::Symbol(loc, core::Names::arg0()), ast::MK::Untyped(loc),
```

Comment:
Just noticed that I haven't updated the signature for the updated method. Two questions:
1. Do I need to? It looks like [this just works](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20A%0A%20%20extend(T%3A%3ASig)%0A%20%20%0A%20%20sig%20%7B%20params(args%3A%20T.untyped).returns(T.untyped)%20%7D%0A%20%20def%20foo(*args%2C%20%26blk)%0A%20%20end%0Aend%0A).
2. If I have to, what is the best way to add a generic block signature?

Reply:
> Do I need to? It looks like this just works.

We've recently discussed that we want to make this be an error, so, yes, it would be nice if you do.

> If I have to, what is the best way to add a generic block signature?

`sig{params(blk: Proc).void}`

Reply:
Ok, I think I was able to add it as `blk: T.nilable(Proc)` (since the block should be optional for these generated methods). Please let me know if I messed something up.